app/web/cypress/e2e/authentication/login.cy.ts

```diff
@@ -1,15 +1,23 @@
-Cypress._.times(import.meta.env.VITE_SI_CYPRESS_MULTIPLIER ? import.meta.env.VITE_SI_CYPRESS_MULTIPLIER : 1, () => {
+const SI_CYPRESS_MULTIPLIER = Cypress.env('VITE_SI_CYPRESS_MULTIPLIER') || import.meta.env.VITE_SI_CYPRESS_MULTIPLIER || 1;
+const AUTH0_USERNAME = Cypress.env('VITE_AUTH0_USERNAME') || import.meta.env.VITE_AUTH0_USERNAME;
+const AUTH0_PASSWORD = Cypress.env('VITE_AUTH0_PASSWORD') || import.meta.env.VITE_AUTH0_PASSWORD;
+const AUTH_API_URL = Cypress.env('VITE_AUTH_API_URL') || import.meta.env.VITE_AUTH_API_URL;
+const SI_WORKSPACE_ID = Cypress.env('VITE_SI_WORKSPACE_ID') || import.meta.env.VITE_SI_WORKSPACE_ID;
+const UUID = Cypress.env('VITE_UUID') || import.meta.env.VITE_UUID || "local";
+
+Cypress._.times(SI_CYPRESS_MULTIPLIER, () => {
   describe("login", () => {
     beforeEach(() => {
       cy.visit("/");
     });
 
     it("log_in", () => {
-      cy.loginToAuth0(import.meta.env.VITE_AUTH0_USERNAME, import.meta.env.VITE_AUTH0_PASSWORD);
-      cy.visit("/");
-      cy.sendPosthogEvent(Cypress.currentTest.titlePath.join("/"), "test_uuid", import.meta.env.VITE_UUID ? import.meta.env.VITE_UUID: "local");
+      cy.loginToAuth0(AUTH0_USERNAME, AUTH0_PASSWORD);
+      cy.visit(AUTH_API_URL + '/workspaces/' + SI_WORKSPACE_ID + '/go');
+      cy.sendPosthogEvent(Cypress.currentTest.titlePath.join("/"), "test_uuid", UUID);
       // check that you're on head i.e. that you were redirected correctly
-      cy.url().should("contain", "head");
+      cy.wait(4000)
```

Comment:
Is this the beginning of the test that ensures SI is ready to go? If so, this will work for now, but it would be great if there was a signal or hook of some kind.

Reply:
Not quite, two things going on here:
1. I believe after the `cy.visit(AUTH_API_URL + '/workspaces/' + SI_WORKSPACE_ID + '/go');` 
 that the /restore_session (or similar) endpoint showed some latency so this was a basic stabilisation workaround to try and make sure the session (and webpage) had been established in the browser before trying to assert that it was on the correct workspace. It would definitely be better to wait for a specific signal from the webpage (like a hook/interception of a specific API call response)
 
2. We did notice some weirdness as to which changeset was loaded into by default by the web app. Currently the application tries to work it out via the session storage keys as to which changeset the user was last on, if the key isn't found it defaults to HEAD. Sometimes in the tests however the last changeset would be B in the session storage key but when the user was routed through to auto it would redirect to HEAD, or vice versa. The check on line 20 was therefore modified to only check it was on the correct workspace, not specifically the correct changeset/HEAD.

Reply:
Gotcha. Feels like for both scenarios we getting around some logic we need to tighten up. However, I think it's unrealistic to chase those down today as the system is rapidly changing, so this little wait works for me!'